### PR TITLE
Update db tables

### PR DESCRIPTION
**Pull request by rshewitt**
# Pull Request

Related to [#4731](https://github.com/GSA/data.gov/issues/4731)

## About

update database tables to reflect what's in the definition document. remove use of sqllite in testing environment because it doesn't support arrays as a dtype which we need for notification_emails in the harvest source table

## PR TASKS

- [x] The actual code changes.
- [ ] Tests written and passed.
- [ ] Any changes to docs?
- [x] Bumped version number in [pyproject.toml](https://github.com/GSA/datagov-harvesting-logic/blob/8078c5b9f015ca7fb8a142e4e4b2e22ad2fd49b1/pyproject.toml#L3) (also checked on [PyPi](https://pypi.org/project/ckanext-geodatagov/#history)).

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -14,15 +14,11 @@
...
-    else:
-        app.config["SQLALCHEMY_DATABASE_URI"] = os.getenv("DATABASE_URI")
-        app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
-        app.config["SECRET_KEY"] = os.urandom(16)
-        Bootstrap(app)
+    app.config["TESTING"] = testing
+    app.config["SQLALCHEMY_DATABASE_URI"] = os.getenv("DATABASE_URI")
+    app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
+    app.config["SECRET_KEY"] = os.urandom(16)
+    Bootstrap(app)
```
Comment (FuhuXia) [inline app/__init__.py]: Do these configs apply to environments other than testing?

Comment (rshewitt) [reply app/__init__.py]: i figure SQLALCHEMY_DATABASE_URI/DATABASE_URI would apply to other environments. everything else can be moved to the [fixture](https://github.com/GSA/datagov-harvesting-logic/blob/03f52c656e0feeff89a9b47ea42fac1b1c3e98d1/tests/conftest.py#L20)

Comment (rshewitt) [reply app/__init__.py]: is `SECRET_KEY` something we want to use?

Comment (FuhuXia) [reply app/__init__.py]: `SECRET_KEY` can anything for the local environment; in cloud.gov it should be a cf env variable.

Comment (btylerburton) [reply app/__init__.py]: set `app.config["TESTING"] = True` in conftest.py and then do conditional logic here based on that.

File: app/__init__.py [hunk 2]
```diff
@@ -14,15 +14,11 @@
...
-    if testing:
-        app.config["TESTING"] = True
-        app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///:memory:"
-        app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
-    else:
-        app.config["SQLALCHEMY_DATABASE_URI"] = os.getenv("DATABASE_URI")
-        app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
-        app.config["SECRET_KEY"] = os.urandom(16)
-        Bootstrap(app)
+    app.config["TESTING"] = testing
```
Comment (btylerburton) [inline app/__init__.py]: we should move this to `conftest.py`

**Review by btylerburton — CHANGES_REQUESTED** [review]
this is good to after you update the `app.config["TESTING"]` line

**Comment by btylerburton** [comment]
Compare is telling me maybe we should just merge this one as it's got the superset of changes

https://github.com/GSA/datagov-harvesting-logic/compare/separate-database...update-db-tables

**Review by btylerburton — DISMISSED** [review]
looks good

**Comment by btylerburton** [comment]
same as other PR, we should get @Jin-Sun-tts or @FuhuXia to review so we get fresh eyes.

**Review by jbrown-xentity — APPROVED** [review]
This looks like it's contained in #72, but looks good!

**Pull request merged**